packages/astro/actions-module.template.mjs

```diff
@@ -0,0 +1,49 @@
+import { ActionError, callSafely } from 'astro:actions';
+
+function toActionProxy(actionCallback = {}, aggregatedPath = '/_actions/') {
+	return new Proxy(actionCallback, {
+		get(target, objKey) {
+			const path = aggregatedPath + objKey.toString();
+			if (objKey in target) {
+				return target[objKey];
+			}
+			async function action(param) {
+				if (import.meta.env.SSR) {
+					throw new ActionError({
+						code: 'BAD_REQUEST',
+						message:
+							'Action unexpectedly called on the server. If this error is unexpected, share your feedback on our RFC discussion: https://github.com/withastro/roadmap/pull/912',
+					});
+				}
+				const headers = new Headers();
+				headers.set('Accept', 'application/json');
+				let body = param;
+				if (!(body instanceof FormData)) {
+					body = param ? JSON.stringify(param) : undefined;
```

Comment:
Is it safe to stringify `param`? Are you sure that it won't fail?

Reply:
Fair, it _might_ fail to stringify. I'll add a try / catch with a readable error.